### PR TITLE
Changes to handle Windows Registry artifact definitions #2429

### PR DESCRIPTION
**Pull request by joachimmetz**
Changes to handle Windows Registry artifact definitions #2429

### CODE REVIEW

File: plaso/engine/artifact_filters.py [hunk 1]
```diff
@@ -65,47 +74,38 @@ def CheckKeyCompatibility(key_path):
         'Prefix of key "{0:s}" is currently not supported'.format(key_path))
     return False
 
-  def BuildFindSpecs(self, environment_variables=None):
+  def BuildFindSpecs(self, artifact_filter_names, environment_variables=None):
```
Comment (Onager) [inline plaso/engine/artifact_filters.py]: We don't have to change it now, but this method isn't needed any more, and the logic can just move to the initializer.

File: plaso/engine/engine.py [hunk 2]
```diff
@@ -317,17 +319,25 @@ def BuildFilterFindSpecs(
...
-      artifact_filters_object.BuildFindSpecs(
-          environment_variables=environment_variables)
-      find_specs = knowledge_base_object.GetValue(
-          artifact_filters_object.KNOWLEDGE_BASE_VALUE)[
-              artifact_types.TYPE_INDICATOR_FILE]
+              artifacts_registry_object, knowledge_base_object))
+      self._artifacts_filter_helper.BuildFindSpecs(
+          artifact_filter_names, environment_variables=environment_variables)
+
+      if self._artifacts_filter_helper.registry_find_specs:
```
Comment (Onager) [inline plaso/engine/engine.py]: Please add a comment here about why this is being done. The idea here is if the user wants a registry artifact, they probably also want to include the registry hive files, right?

Also, doesn't this logic belong in the artifact filter helper, given this is about constructing artifact filters? Please move it there.

File: plaso/parsers/mediator.py [hunk 3]
```diff
@@ -22,26 +22,32 @@ class ParserMediator(object):
...
+    artifacts_filter_helper (ArtifactDefinitionsFilterHelper): artifacts
+        definitions filter helper.
     last_activity_timestamp (int): timestamp received that indicates the last
         time activity was observed. The last activity timestamp is updated
         when the mediator produces an attribute container, such as an event
         source. This timestamp is used by the multi processing worker process
         to indicate the last time the worker was known to be active. This
         information is then used by the foreman to detect workers that are
         not responding (stalled).
+    registry_find_specs (list[dfwinreg.FindSpec]): Windows Registry find
```
Comment (Onager) [inline plaso/parsers/mediator.py]: No such attribute, please remove.

File: plaso/engine/artifact_filters.py (hunk 1 above)
Comment (joachimmetz) [reply plaso/engine/artifact_filters.py]: Let's chat about this, it not entirely clear to me what you mean.

File: plaso/engine/engine.py (hunk 2 above)
Comment (joachimmetz) [reply plaso/engine/engine.py]: Let's keep it here for now, and have a longer conversation about this. I'll add the comment.

File: plaso/parsers/mediator.py (hunk 3 above)
Comment (joachimmetz) [reply plaso/parsers/mediator.py]: ack, removed

File: plaso/engine/engine.py (hunk 2 above)
Comment (Onager) [reply plaso/engine/engine.py]: OK, https://github.com/log2timeline/plaso/issues/2438

File: plaso/engine/artifact_filters.py (hunk 1 above)
Comment (Onager) [reply plaso/engine/artifact_filters.py]: We can just pass artifact_filter_names and environment in __init__ and we don't need this method. In the engine, which is the only place ArtifactDefinitionsFilterHelper is initialized outside of tests, we call the methods directly after each other. 
Filed https://github.com/log2timeline/plaso/issues/2438 for this (and other changes).

**Pull request merged**